go/arrow/csv/transformer.go

```diff
@@ -215,6 +216,15 @@ func (w *Writer) transformColToStringArr(typ arrow.DataType, col arrow.Array) []
 				res[i] = w.nullValue
 			}
 		}
+	case arrow.BinaryDataType:
```

Comment:
same as above, make sure that `LargeString` is covered above this case so it doesn't fall into here.